### ISSUE TITLE
Error: Unexpected content type in insert operation

### ISSUE THREAD

**Issue by iSplasher** [BitPhinix/slate-yjs#119]
Hello, 
I'm trying to set up a simple project following the demo but this line of code is failing with **Error: Unexpected content type in insert operation**
```js
syncDoc.insert(0, doc.map(toSyncElement));
```

Here's my default value
```
const defaultValue = [
  {
    type: 'paragraph',
    children: [
      {
        text: '',
      },
    ],
  },
];
```

Let me know if you need anything else.

Using latest slate-js and latest yjs

**Comment by iSplasher** [comment]
I did some debugging, and it seems like deep inside Yjs' insert operation: 
```js
const typeListInsertGenericsAfter = (transaction, parent, referenceItem, content) => {
  ...
  content.forEach(c => {
    switch (c.constructor) {
      case Number:
      case Object:
      case Boolean:
      case Array:
      case String:
        jsonContent.push(c);
        break
      default:
        packJsonContent();
        switch (c.constructor) {
          case Uint8Array:
          case ArrayBuffer:
            left = new Item(createID(ownClientId, getState(store, ownClientId)), left, left && left.lastId, right, right && right.id, parent, null, new ContentBinary(new Uint8Array(/** @type {Uint8Array} */ (c))));
            left.integrate(transaction, 0);
            break
          case Doc:
            left = new Item(createID(ownClientId, getState(store, ownClientId)), left, left && left.lastId, right, right && right.id, parent, null, new ContentDoc(/** @type {Doc} */ (c)));
            left.integrate(transaction, 0);
            break
          default:
            if (c instanceof AbstractType) {
              left = new Item(createID(ownClientId, getState(store, ownClientId)), left, left && left.lastId, right, right && right.id, parent, null, new ContentType(c));
              left.integrate(transaction, 0);
            } else {
              throw new Error('Unexpected content type in insert operation')
            }
        }
    }
  });
  packJsonContent();
};
```
This part evaluates to false `if (c instanceof AbstractType) {` even though `c` should be a YMap which extends AbstractType. Not really sure what's going on here.

**Comment by iSplasher** [comment]
A simpler test:
```js
import { toSyncDoc, toSyncElement } from 'slate-yjs';
import * as Y from 'yjs';

const content= [
  {
    type: 'paragraph',
    children: [
      {
        text: '',
      },
    ],
  },
];

const YContent = content.map(toSyncElement)[0]; // is YMap
YContent instanceof Y.Map // evaluates to false
```

Again, not sure why it would fail like that. Maybe it's about mixing commonjs and esm modules? I really don't know.

**Comment by iSplasher** [comment]
I ended up copying the src folder directly into my project so it could be transpiled the same way my own code is, and it worked. So i'm guessing it has to do with something babel or webpack is doing to cause it to fail like that. Not really sure what you as a library author can do against that, so i'll close the issue.

**Comment by BitPhinix** [comment]
Hi @iSplasher,

Sorry for the late reply. I did some digging into this and the easy way to fix it is to add yjs as a peer dependency and not as a normal one. I'll create a pr for it this weekend (really busy right now)

**Comment by BitPhinix** [comment]
Should be fixed in v1.0.0
